editor/src/clj/editor/workspace.clj

```diff
@@ -291,18 +292,31 @@ ordinary paths."
 (defn- absolute-path [^String path]
   (.startsWith path "/"))
 
+(defn- get-parent-dir [^String path]
+  (str (.getParent (File. path))))
+
+(defn- relative-path [^String base ^String path]
+  (let [path-parts (str/split path #"\.\.\/")
+        parent-path (if (= "/" base)
+                      base
+                      (str base "/"))]
+    (prn parent-path)
+    (if (> (count path-parts) 1)
+      (recur (get-parent-dir parent-path) (string/join "/" (drop 1 path-parts)))
+      (str parent-path path))))
+
 (defn to-absolute-path
   ([rel-path] (to-absolute-path "" rel-path))
   ([base rel-path]
    (if (absolute-path rel-path)
      rel-path
-     (str base "/" rel-path))))
+     (relative-path base rel-path))))
 
 (defn resolve-resource [base-resource path]
   (when-not (empty? path)
     (let [path (if (absolute-path path)
                  path
-                 (to-absolute-path (str (.getParent (File. (resource/proj-path base-resource)))) path))]
+                 (to-absolute-path (get-parent-dir (resource/proj-path base-resource)) path))]
```

Comment:
Our implementation of `to-absolute-path` looks weird to me. I suggest we just revert your changes and bypass it. I think this line should make use of the built-in `File` methods instead, i.e:
```clj
(.getCanonicalPath (io/file (.getParentFile (io/file base-resource)) path))
```
The `getCanonicalPath` method can deal with a lot more complex scenarios than our primitive string mangling. Simple example:
```clj
(.getCanonicalPath (io/file (.getParentFile (io/file "/mats/project-home/subdirectory/shader.vp"))
                            "../include.glsl"))
=> "/mats/project-home/include.glsl"
```

Or this complex, horrifying example resolving `".././subdirectory/../include.glsl"`:
```clj
(.getCanonicalPath (io/file (.getParentFile (io/file "/mats/project-home/subdirectory/shader.vp"))
                            ".././subdirectory/../include.glsl"))
=> "/mats/project-home/include.glsl"
```